modules/standard/Random.chpl

```diff
@@ -324,20 +328,21 @@ module Random {
 
   pragma "no doc"
   /* _choice branch for distribution defined by probabilities array */
-  proc _choiceProbabilities(stream, arr:[], size:?sizeType, replace, prob:?probType) throws
+  proc _choiceProbabilities(stream, X:domain, stride:int, low:int, size:?sizeType, replace, prob:?probType) throws
   {
     import Search;
     import Sort;
 
     // If stride, offset, or size don't match, we're in trouble
-    if arr.domain != prob.domain then
-      throw new owned IllegalArgumentError('choice() arrays must have equal domains');
+    /*
+    if X != prob.domain then
```

Comment:
How should we handle this, as the original array is reindexed in the overload function itself?

Reply:
I think it's OK for stride or offset to not match for the domain/range case.

Reply:
thanks. I have removed the check